packages/auth-client/src/controllers/engine.ts

```diff
@@ -381,7 +382,7 @@ export class AuthEngine extends IAuthEngine {
       await this.client.core.pairing.activate({ topic: pairingTopic });
 
       const { s: signature, p: payload } = response.result;
-      await this.client.requests.set(id, { id, ...response.result });
+      await this.client.requests.set(id, { id, topic, ...response.result });
```

Comment:
This is writing the per-request response topic (which is different to the static pairing topic the dapp/wallet share) instead of the pairingTopic to `requests`:

```suggestion
      await this.client.requests.set(id, { id, pairingTopic, ...response.result });
```